engine/step/hook.go

```diff
@@ -0,0 +1,27 @@
+package step
+
+import (
+	"github.com/juju/errors"
+)
+
+// Hook represents the hook data structure
+type Hook struct {
+	Name            string            `json:"name"`
+	Description     string            `json:"description""`
+	LongDescription *string           `json:"long_description,omitempty"`
+	DocLink         *string           `json:"doc_link,omitempty"`
+	Action          Executor          `json:"action"`
+	Results         map[string]string `json:"results"`
```

Comment:
Also, maybe we should keep this singular as `models/task/task.go`

```suggestion
	Result         map[string]interface{} `json:"result"`
```